Documentation/ApiOverview/FormEngine/Rendering/Index.rst

```diff
@@ -83,60 +71,39 @@ including containers, elements, fieldInformation, fieldWizards and fieldControls
+    :language: php
+    :caption: EXT:my_extension/ext_localconf.php
 
 This re-routes the :php:`renderType` "flex" to an own class. If multiple registrations for a single renderType exist,
 the one with highest priority wins.
 
-.. note::
-   The :php:`NodeFactory` uses :php:`$data['renderType']`.
-   A couple of TCA fields actively use this renderType. However, it is important to understand the renderType is *only*
-   used within the FormEngine and :php:`type` is still a must-have setting for columns fields in TCA. Additionally,
-   :php:`type` can *not* be overridden in :php:`columnsOverrides`. Basically, :php:`type` specifies how the DataHandler
-   should put data into the database, while :php:`renderType` specifies how a single field is rendered. This additionally
-   means there can exist multiple different renderTypes for a single type, and it means it is possible to invent a new
-   renderType to render a single field differently, but still let the DataHandler persist it the usual way.
+..  note::
+    The :php:`NodeFactory` uses :php:`$data['renderType']`.
+    A couple of TCA fields actively use this renderType. However, it is important to understand the renderType is *only*
+    used within the FormEngine and :php:`type` is still a must-have setting for columns fields in TCA. Additionally,
+    :php:`type` can *not* be overridden in :php:`columnsOverrides`. Basically, :php:`type` specifies how the DataHandler
+    should put data into the database, while :php:`renderType` specifies how a single field is rendered. This additionally
+    means there can exist multiple different renderTypes for a single type, and it means it is possible to invent a new
+    renderType to render a single field differently, but still let the DataHandler persist it the usual way.
 
 
 Adding a new renderType in :file:`ext_localconf.php`
 
-.. code-block:: php
-
-    // Add new field type to NodeFactory
-    $GLOBALS['TYPO3_CONF_VARS']['SYS']['formEngine']['nodeRegistry'][1487112284] = [
-        'nodeName' => 'selectTagCloud',
-        'priority' => '70',
-        'class' => \MyVendor\CoolTagCloud\Form\Element\SelectTagCloudElement::class,
-    ];
+..  literalinclude:: _ext_localconf.php
+    :language: php
+    :caption: EXT:my_extension/ext_localconf.php
 
 And use it in TCA for a specific field, keeping the full database functionality in DataHandler together with the
 data preparation of FormDataCompiler, but just routing the rendering of that field to the new element:
 
-.. code-block:: php
-
-    $GLOBALS['TCA']['myTable']['columns']['myField'] = [
-        'label' => 'Cool Tag cloud',
-        'config' => [
-            'type' => 'select',
-            'renderType' => 'selectTagCloud',
-            'foreign_table' => 'tx_cooltagcloud_availableTags',
-        ],
-    ];
-
+..  literalinclude:: _tx_cooltagcloud.php
+    :language: php
+    :caption: EXT:cool_tag_cloud/Configuration/TCA/overrides/tx_cooltagcloud.php
```

Comment:
What is cool_tag_cloud? The extension does not exist on TER? I am fine with adjusting it in a separate PR.